util/util.go

```diff
@@ -1,9 +1,37 @@
 package util
 
-import "os/exec"
+import (
+	"errors"
+	"net"
+	"os/exec"
+	"strings"
+)
 
 // IsCommandAvailable check command exits.
 func IsCommandAvailable(cmd string) bool {
 	_, err := exec.LookPath(cmd)
 	return err == nil
 }
+
+func GetClientIp() (string, error) {
+	addresses, err := net.InterfaceAddrs()
+
+	if err != nil {
+		return "", err
+	}
+
+	for _, address := range addresses {
+		if ipnet, ok := address.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
+			if ipnet.IP.To4() != nil {
+				return ipnet.IP.String(), nil
+			}
+
+		}
+	}
+
+	return "", errors.New("can not find the client ip addresses")
+}
+
+func CountWords(s string) int {
+	return len(strings.Fields(s))
+}
```

Comment:
Error generating comment for file util/util.go: Post "https://api.openai.com/v1/chat/completions": proxyconnect tcp: dial tcp 127.0.0.1:10838: connectex: No connection could be made because the target machine actively refused it.